MediaBrowser.Model/Entities/MediaStream.cs

```diff
@@ -112,7 +112,15 @@ public string DisplayTitle
                         attributes.Add("Default");
                     }
 
-                    return string.Join(" ", attributes);
+                    if (!string.IsNullOrEmpty(Title)) {
+                        return attributes.AsEnumerable()
+                        // keep Tags that are not already in Title
+                        .Where(tag => Title.IndexOf(tag, StringComparison.OrdinalIgnoreCase) == -1).Aggregate(new StringBuilder(Title), (builder, attr) => builder.Append(" - ").Append(attr))
+                        // attributes concatenation, starting with Title
+                        .ToString();
```

Comment:
A few notes:
* I think `AsEnumerable()` is not necessary since `ToList()` is already enumerable
* All the chained method calls below `return attributes` should be indented one level
* Put `.Aggregate(...)` on a separate line to make this more readable
* Put the comment for the `Aggregate()` function above the function call instead of below it
* There is a commented-out `if (!string.IsNullOrEmpty(Title))` code block [above](https://github.com/jellyfin/jellyfin/pull/2773/files#diff-38cc4f9175bd3c3da5251a3d7cab56beR82) that should be deleted

Reply:
All good points, updated!